src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -20,7 +20,7 @@ export interface UiSettingsServiceOptions {
   type: string;
   id: string;
   buildNum: number;
-  savedObjectsClient: SavedObjectsClientContract;
+  savedObjectsClient?: SavedObjectsClientContract;
```

Comment:
I agree, but couldn't find a better way yet. I tried to propagate just result of `getRegistered` call instead of an entire client, but it required a way too many changes in the `RenderingService` code.

I'll give it another thought though.

Reply:
Handled in https://github.com/elastic/kibana/pull/103636/commits/c3b8ede0ec9247dfcd8a90003451314814fae1bb